llm_github_models.py

```diff
@@ -255,3 +255,37 @@ def execute(
             )
             response.response_json = None  # TODO
             yield completion.choices[0].message.content
+
+
+class GitHubEmbeddingModel(llm.EmbeddingModel):
+    needs_key = "github"
+    key_env_var = "GITHUB_MODELS_KEY"
+    batch_size = 100
+
+    def __init__(self, model_id: str, dimensions: Optional[int] = None):
+        self.model_id = f"github/{model_id}"
+        if dimensions is not None:
+            self.model_id += f"-{dimensions}"
+
+        self.model_name = model_id
+        self.dimensions = dimensions
+
+    def embed_batch(self, texts: List[str]) -> Iterator[List[float]]:
+        if not texts:
+            return []
+
+        key = self.get_key()
+        client = EmbeddingsClient(
+            endpoint=INFERENCE_ENDPOINT,
+            credential=AzureKeyCredential(key),
+        )
+
+        kwargs = {
+            "input": texts,
```

Comment:
input in EmbeddingModel accepts a iterable of str or bytes, whereas `client.embed` takes str or Embedding Token. For now, I think bytes will probably fail. We should add a test for that too